### PR TITLE
Bug 1793850: Update UsingDeprecatedAPIExtensionsV1Beta1 alert

### PR DESCRIPTION
**Pull request by soltysh**
/assign @tnozicka

### CODE REVIEW

File: manifests/0000_90_kube-apiserver-operator_04_servicemonitor-apiserver.yaml [hunk 1]
```diff
@@ -110,8 +110,16 @@ spec:
...
       expr: |
-        apiserver_request_count{group="extensions",version="v1beta1",resource!~"ingresses|",client!~"hyperkube/.*|cluster-policy-controller/.*"}
+        increase(
+          apiserver_request_total{
+            group="extensions",
+            version="v1beta1",
+          }[24h]
+        ) > 0
       labels:
         severity: warning
```
Comment (cblecker) [inline manifests/0000_90_kube-apiserver-operator_04_servicemonitor-apiserver.yaml]: Should this be a warning still?

File: manifests/0000_90_kube-apiserver-operator_04_servicemonitor-apiserver.yaml [hunk 2]
```diff
@@ -110,8 +110,16 @@ spec:
     rules:
     - alert: UsingDeprecatedAPIExtensionsV1Beta1
       annotations:
-        message: A client in the cluster is using deprecated extensions/v1beta1 API that will be removed soon.
+        summary: "A client is using a deprecated {{"{{"}} $labels.group {{"}}"}}/{{"{{"}} $labels.version {{"}}"}} API version"
+        message: |
+          A client in the cluster has used a deprecated {{"{{"}} $labels.group {{"}}"}}/{{"{{"}} $labels.version {{"}}"}} API version of the {{"{{"}} $labels.resource {{"}}"}} resource in the last 24 hours.
+          This API version may be removed soon; please migrate to the current version of this API.
       expr: |
-        apiserver_request_count{group="extensions",version="v1beta1",resource!~"ingresses|",client!~"hyperkube/.*|cluster-policy-controller/.*"}
```
Comment (cblecker) [inline manifests/0000_90_kube-apiserver-operator_04_servicemonitor-apiserver.yaml]: You're losing the exception in here for the `velero-server` which, as a piece of backup software, will scrape every api endpoint advertised in the system. Are these APIs completely removed in 4.4+, and no longer advertised?

File: manifests/0000_90_kube-apiserver-operator_04_servicemonitor-apiserver.yaml (hunk 1 above)
Comment (soltysh) [reply manifests/0000_90_kube-apiserver-operator_04_servicemonitor-apiserver.yaml]: Yes.

File: manifests/0000_90_kube-apiserver-operator_04_servicemonitor-apiserver.yaml (hunk 2 above)
Comment (soltysh) [reply manifests/0000_90_kube-apiserver-operator_04_servicemonitor-apiserver.yaml]: I'd prefer we add them if we need them. I don't see any reason to add it, yet.

**Comment by crawford** [comment]
/hold

Something is [wrong](https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_cluster-kube-apiserver-operator/742/pull-ci-openshift-cluster-kube-apiserver-operator-master-e2e-aws/3426/artifacts/e2e-aws/pods/openshift-cluster-version_cluster-version-operator-89dc78b79-ht8nj_cluster-version-operator.log) with your manifest:

```
error parsing 0000_90_kube-apiserver-operator_04_servicemonitor-apiserver.yaml: error parsing: error converting YAML to JSON: yaml: line 13: did not find expected key
```

**Comment by cblecker** [comment]
@soltysh I'm not sure what's up with this. The error @crawford mentioned is consistent, but we don't modify L13 in this file.. and other PRs are merging.

File: manifests/0000_90_kube-apiserver-operator_04_servicemonitor-apiserver.yaml [hunk 3]
```diff
@@ -110,8 +110,16 @@ spec:
     rules:
     - alert: UsingDeprecatedAPIExtensionsV1Beta1
       annotations:
-        message: A client in the cluster is using deprecated extensions/v1beta1 API that will be removed soon.
+        summary: "A client is using a deprecated {{"{{"}} $labels.group {{"}}"}}/{{"{{"}} $labels.version {{"}}"}} API version"
```
Comment (LiliC) [inline manifests/0000_90_kube-apiserver-operator_04_servicemonitor-apiserver.yaml]: Didn't we say we don't need this alert anymore as the client can't use this extensions/v1beta1 anyways? 🤔

**Comment by LiliC** [comment]
In general regarding these alerts about deprecated APIs my two cents:
They are really broken in 4.3, we have a long running cluster and someone created a DS with deprecated API and even after removal the alert has not gone away, because it does not use an increase. We silenced them, but they still show in the main console, so right now users can't get rid of these alerts. This needs to be fixed **ASAP** in 4.3 onwards. This is inducing alert fatique for users if we do not fix them, and making users ignore alerts in general.

tl;dr:
- one alert for user namespaces, warning or info level is fine
- lets group the clients into one, I don't want to see 4 alerts firing if someone creates one DS.
- no alerts for openshift components only we can control, you can add a recording rule test in origin for this to prevent new merges for happening. Users can't influence this nor fix it, so its not an actionable alert. 

Please ping me or @openshift/openshift-team-monitoring for any help, or advice, we should fix this soon. Thanks! 

(It seems we have like 3 PRs open for this, forgot which was the original one I was pinged on 🤔 )

**Comment by cblecker** [comment]
Looks like CI is good now. Pulling the hold that @crawford put in [here](https://github.com/openshift/cluster-kube-apiserver-operator/pull/742#issuecomment-580358311) to stop the retest loop.

/hold cancel

**Comment by s-urbaniak** [comment]
@soltysh @cblecker It is a bummer that this got merged despite the comment in https://github.com/openshift/cluster-kube-apiserver-operator/pull/742#issuecomment-584089082 :-( As @LiliC mentioned this will cause alert fatigue and should rather be catched in e2e tests.

@LiliC will submit a follow-up PR, or do you have concrete follow-up plans here?

**Comment by soltysh** [comment]
@s-urbaniak @LiliC sorry, this got merged w/o my notice, fix excluding openshift elements in https://github.com/openshift/cluster-kube-apiserver-operator/pull/762
